addons/sale_timesheet/wizard/sale_make_invoice_advance_views.xml

```diff
@@ -15,11 +15,14 @@
                     name="timesheet_invoice_date_range"
                     attrs="{'invisible': ['|', ('invoicing_timesheet_enabled', '=', False), ('advance_payment_method', '!=', 'delivered')]}"
                 >
+                    <label for="date_start_invoice_timesheet" string="Timesheets Period" attrs="{'invisible': [ '|', ('invoicing_timesheet_enabled', '=', False), ('advance_payment_method', '!=', 'delivered')]}"/>
                     <field
                         name="date_start_invoice_timesheet"
                         widget="daterange"
+                        attrs="{'required': ['|', ('date_start_invoice_timesheet', '!=', False), ('date_end_invoice_timesheet', '!=', False)]}"
```

Comment:
is this really needed ? can't you just do this ?
```suggestion
                        required="1"
```

Reply:
@pkyriakou we cannot because the daterange should be only required when there is at least one date set. If the both fields are empty then the daterange should not be required. 🙂

I agree it is a bit strange to write the required condition like this but it does not seem we have another way to do that. 😅